src/Common/ProcessExecutor.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Robo\Common;
+
+use Robo\Contract\OutputAdapterInterface;
+use Robo\Contract\OutputAwareInterface;
+use Robo\Contract\TaskInterface;
+use Robo\Contract\VerbosityThresholdInterface;
+use Robo\Robo;
+use Robo\Task\BaseTask;
+
+class ProcessExecutor implements OutputAwareInterface
+{
+    use ExecTrait;
+    use TaskIO; // uses LoggerAwareTrait and ConfigAwareTrait
+    use ProgressIndicatorAwareTrait;
+    use OutputAwareTrait;
+
+    public function __construct($process)
+    {
+        $this->process = $process;
+        $this->logger = Robo::logger();
+        $this->setOutputAdapter(new OutputAdapter());
+        $this->outputAdapter()->setOutput(Robo::output());
```

Comment:
@DavertMik Do you concur with the strategy described ^^? LMK if you have a different perspective. If we wanted to be really strict, we could make the logger and output interface constructor parameters required, and force users to utilize `Robo::logger()` and `Robo::output()`, if that is what's desired.

Perhaps a middle-ground in terms of strictness would be to make all of the constructor parameters required, and then provide a static factory method (`ProcessExecutor::create()`, or `Robo::process()`) that supplies the logger and output interface via the static service methods. There's some nice symmetry to that.